### DISCUSSION TITLE
DialogHost.Show() with a specific style

### DISCUSSION THREAD

**Discussion by ElieTaillard** [MaterialDesignInXAML/MaterialDesignInXamlToolkit#3114] [Q&A] [answered]
I want to open a dialog from a ViewModel (like the sample 3 in the demo application) with the `EmbeddedDialogHostStyle`.

Is it possible to call `DialogHost.Show()` specifying a style to use?

**Comment by nicolaihenriksen** [comment, marked as answer]
@Xaalek I don't think that is currently possible. You statically define this with `Style="{StaticResource ...}"` directly on the `DialogHost` element.

So I guess what you're requesting is to specify a "temporary" style to use for this particular call to `DialogHost.Show()`? 

I guess it would be possible to introduce a scheme to temporarily inject a style. This could either be in the `DialogHost.Show()` overloads, or perhaps you need a utility method which will find the relevant `DialogHost` and apply a given style on it (which you could call from your ViewModel - although that smells a little; but so does calling `DialogHost.Show()` from your VM one could argue).

There may be some challenges with "when to restore the original style". Is this something you as a caller need to do? Or should it happen when `IsOpen` changes to false...

**Reply by ElieTaillard** [reply]
Yes, I hadn't thought of that. Ultimately, I would like all my Dialogs to be embedded. Do you think that making the EmbeddedDialogHostStyle the default style for the Dialog is a solution that could work?

**Reply by nicolaihenriksen** [reply]
Couldn't you just apply the embedded style on a DialogHost placed at the root Window(s) of your application then?

**Reply by Keboo** [reply]
I am a little hesitant to change the default as the embedded dialogs are not able to exceed the size of the parent control (window). However, for many apps I could see that default being problematic. I think @nicolaihenriksen is correct, the simplest option for changing the default is to put it an implicit style (at a top level; window, or app.xaml). 

```xaml
<Style TargetType="{x:Type wpf:DialogHost}" BasedOn="{StaticResource MaterialDesignEmbeddedDialogHost}" />
```

**Reply by ElieTaillard** [reply]
Thank you for your answers. Yeah that's what I did and it actually works. Unfortunately, I noticed that when using the MaterialDesignEmbeddedDialogHost I had a drop in performance. So in the end I think I'll keep the classic style for this reason.

**Reply by nicolaihenriksen** [reply]
> Unfortunately, I noticed that when using the MaterialDesignEmbeddedDialogHost I had a drop in performance.

Can you elaborate a bit on that? I am interested in understanding why/how that affects perf.

**Reply by ElieTaillard** [reply]
Popup opening animations are less fluid with the MaterialDesignEmbeddedDialogHost style. And I had my Snoop crashing with this style. For your information, I use a MaterialDesign fork so I don't have the latest version of the code and the code is altered. Maybe with the last version of the library there is not this problem
